diff --git a/node_modules/codegen-graph-ts/build/src/gen/methods/react-query.js b/node_modules/codegen-graph-ts/build/src/gen/methods/react-query.js
index 7985cb7..a8516e8 100644
--- a/node_modules/codegen-graph-ts/build/src/gen/methods/react-query.js
+++ b/node_modules/codegen-graph-ts/build/src/gen/methods/react-query.js
@@ -19,7 +19,7 @@ function reactquery(schema) {
             const enabled = options && args;
 
             return useQuery(
-                ['codegen-graphql', enabled ? generateGql('${entity.name}', options, args) : null],
+                ['codegen-graphql', enabled ? generateGql('${entity.name}', options, args) : null, queryOptions?.queryKey?.toString()],
                 async () => func(url, options!, args!),
                 {
                     ...queryOptions,
@@ -33,7 +33,7 @@ function reactquery(schema) {
             const enabled = options && args;
 
             return useQuery(
-                ['codegen-graphql', enabled ? generateGql('${entity.name}s', options, args) : null],
+                ['codegen-graphql', enabled ? generateGql('${entity.name}s', options, args) : null, queryOptions?.queryKey?.toString()],
                 async () => func(url, options!, args!),
                 {
                     ...queryOptions,
